ludwig/datasets/loaders/av_mnist.py

```diff
@@ -0,0 +1,214 @@
+
+                self.audio_data = self.audio_data[:, np.newaxis, :, :]
+                self.mnist_data = self.mnist_data.reshape(self.mnist_data.shape[0], 1, 28, 28)
+                self.write_output_dataset(self.labels, self.audio_data, os.path.join(self.raw_dataset_dir, dataset))
+                self.write_output_dataset(self.labels, self.mnist_data, os.path.join(self.raw_dataset_dir, dataset))
+            else:
+                if self.modal:
+                    if self.modal not in ["audio", "image"]:
+                        raise ValueError("the value of modal is allowed")
+                    if dataset == "train":
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "train_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                    else:
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "test_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                    if self.modal == "audio":
+                        self.data = self.data[:, np.newaxis, :, :]
+                    elif self.modal == "image":
+                        self.data = self.data.reshape(self.data.shape[0], 1, 28, 28)
+                    self.write_output_dataset(self.labels, self.data, os.path.join(self.raw_dataset_dir, dataset))
+                else:
+                    raise ValueError("the value of modal should be given")
+
+        return super().transform_files(file_paths)
+
+    def load_unprocessed_dataframe(self, file_paths: List[str]) -> pd.DataFrame:
+        """Load dataset files into a dataframe."""
+        return self.output_training_and_test_data()
+
+    def process_source_dataset(self, raw_dataset_dir="."):
+        """An intermediate method to write .npy files for the training and the test image and label data to the raw dataset directory
+        :args:
+            raw_dataset_dir (str) : the directory where all the raw data lives
+        :returns:
+            None"""
+        file_names = {
+            "train_data": "train-images-idx3-ubyte.gz",
+            "train_labels": "train-labels-idx1-ubyte.gz",
+            "test_data": "t10k-images-idx3-ubyte.gz",
+            "test_labels": "t10k-labels-idx1-ubyte.gz",
+        }
+        print("Raw dataset working directory: %s" % raw_dataset_dir)
+
+        for key, file_name in self.file_names.items():
+            file_path = os.path.join(raw_dataset_dir, file_name)
+            print("file: %s" % key)
+            f = open(file_path)
+            # read the definition of idx1-ubyte and idx3-ubyte
+            f.seek(4)
+            num = f.read(4)
+            num = int().from_bytes(num, "big")
+            print("size of %s : %d" % (key, num))
+            if re.match(r".*data.*", key) is not None:
```

Comment:
nit: perhaps, it may be cleaner to use something like:

```
if 'data' in key:
    ....
```

Either is fine!

Reply:
Added